### PR TITLE
Using bitmap API from CoreRustBoy

### PR DESCRIPTION
**Pull request by SeanAsby**

### CODE REVIEW

File: RustBoy/RustBoy.swift [hunk 1]
```diff
@@ -7,103 +7,147 @@
...
+    }
 
 	internal init() {}
 
-	internal func buttonDown(_ button: ButtonType) {
-		guard let coreRef = coreRef else { return }
+    internal func buttonDown(_ button: ButtonType) {
 #if os(OSX)
-		rustBoyButtonClickDown(coreRef, button.asCoreButton)
+        coreRustBoy?.buttonDown(rustBoyButton(button))
```
Comment (coopersimon) [inline RustBoy/RustBoy.swift]: I kinda feel this should be buttonPressed and buttonUnpressed or something? buttonDown can become confused with the Down button.

File: RustBoy/RustBoy.swift [hunk 2]
```diff
@@ -7,103 +7,147 @@
...
+    fileprivate init?(cartridge: RustBoy.Cartridge) {
+        guard let coreRef = rustBoyCreate(cartridge.path, cartridge.saveFilePath) else { return nil }
+        self.coreRef = coreRef
+        timer = Timer.scheduledTimer(withTimeInterval: 1 / Self.framerate, repeats: true) { timer in
+            self.render()
+        }
+    }
+
+    fileprivate func buttonDown(_ button: rustBoyButton) {
+        rustBoyButtonClickDown(coreRef, button)
```
Comment (coopersimon) [inline RustBoy/RustBoy.swift:92]: Again, maybe this should be rustBoyButtonPressed and Unpressed

File: super_rust_boy_c_binding/src/lib.rs [hunk 3]
```diff
@@ -79,6 +76,17 @@ pub unsafe extern fn rustBoyDelete(instance: *const c_void) {
     rust_boy.drop_in_place();
 }
 
+#[no_mangle]
+pub unsafe extern fn rustBoyFrame(instance: *const c_void, buffer: *mut u32, length: u32) {
+	let rust_boy = instance as *mut RustBoy;
+    if let Some(rust_boy_ref) = rust_boy.as_mut() {
```
Comment (coopersimon) [inline super_rust_boy_c_binding/src/lib.rs:82]: Weird mix of spaces and tabs here?

Comment (SeanAsby) [reply super_rust_boy_c_binding/src/lib.rs:82]: Fixed

**Review by coopersimon — APPROVED** [review]

**Pull request merged**